lib/net/ldap/connection.rb

```diff
@@ -31,10 +36,15 @@ def open_connection(server)
     hosts = server[:hosts]
     encryption = server[:encryption]
 
+    socket_opts = {
+      connect_timeout: server[:connect_timeout] || DefaultConnectTimeout,
+      timeout: server[:timeout] || DefaultTimeout
```

Comment:
Interesting, it was there in 2.0.0... http://docs.ruby-lang.org/en/2.0.0/Socket.html#method-c-tcp
I'm gonna remove it :smile: